src/constructorio-id.js

```diff
@@ -245,20 +246,29 @@
     this.session_id = sessionId;
     this.session_is_new = sessionData && sessionDataId === sessionId ? false : true;
 
+    // persist new status for when ciojs-client is instantiated before beacon
+    if (sessionData && sessionData.newToBeacon) {
+      this.new_to_beacon = true;
+    }
+
+    const storedData = {
+      sessionId: sessionId,
+      lastTime: now
+    };
+
+    // persist new status for when ciojs-client is instantiated before beacon
+    if (this.session_is_new) {
+      storedData.newToBeacon = true;
```

Comment:
With the current logic, this would set `newToBeacon` to true and write to storage for the first time. Since we never set it to false after that, wouldn't it always stay true 🤔 ? Or should we set it to false in the beacon 🤔 I'm a bit confused

Reply:
This should run off of the current `session_is_new` logic, so if the session is no longer new, it shouldn't be set since the storedData obj gets instantiated every time I think